### PR TITLE
Remove inlay hint in diff views

### PR DESCRIPTION
**Pull request by vbfox**
If the left side of a diff view that contain the old version of the file apply inlays they are misplaced and produce a weird display:

image

After the change:

image

The detection is done by blacklisting the url schemes used by git and subversion scm extensions, whitelisting `file` is also possible but neither is perfect as VSCode now support both pluggable scm extensions and pluggable remote filesystems. But I suspect that the list of scm extensions is more easily manageable.

**Note**: I can rebase on #3378 if needed as it touches the same lines of code

### CODE REVIEW

File: editors/code/src/inlay_hints.ts [hunk 1]
```diff
@@ -90,7 +92,14 @@ class HintsUpdater {
 
     private get allEditors(): vscode.TextEditor[] {
         return vscode.window.visibleTextEditors.filter(
-            editor => editor.document.languageId === 'rust',
+            editor => {
+                if (editor.document.languageId !== 'rust') {
```
Comment (lnicola) [inline editors/code/src/inlay_hints.ts]: My slight preference would be for `editor => editor.document.languageId && editor.document.uri.scheme !== 'git' && editor.document.uri.scheme !== 'svn'`.

**Comment by vbfox** [comment]
Reading a little bit more code I think i'll also change & share code with `Ctx.activeRustEditor` as all the callers assume that the document with the URI is it's current version in the LSP server but that's not the case for scm diff views so it'll cause the same class of problems.

**Comment by matklad** [comment]
Yeah, makes sense. I wonder if we should maybe also lift `visibleRustEditors` to `Ctx`.

**Comment by vbfox** [comment]
Done, I also made the check a separate function in `utils` and called it everywhere the `languageId` was checked.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**